node/node.go

```diff
@@ -77,3 +78,17 @@ func (n *Node) GetLogger() log.Logger {
 func (n *Node) EventBus() *types.EventBus {
 	panic("implement me")
 }
+
+func (n *Node) loadPrivateKey(nodeKey p2p.NodeKey) error {
+	switch nodeKey.PrivKey.Type() {
+	case "ed25519":
+		privKey, err := crypto.UnmarshalEd25519PrivateKey(nodeKey.PrivKey.Bytes())
```

Comment:
We should at least consider if doesn't make more sense, to directly generate a libp2p/crypto key in `startInProcess` and pass it into optimint directly instead. It seems like we have to modify that file (server/start.go) in the sdk anyway. For now that shouldn't matter to much until we have a clearer picture of the other changes in the sdk (if any).

Reply:
Very good comment. I was focusing on minimizing changes in cosmos-sdk, but probably I went to far.
Created tracking issue for this: #15.